src/Compilers/Core/Portable/CommandLine/AnalyzerConfigSet.cs

```diff
@@ -95,8 +95,10 @@ public AnalyzerConfigOptionsResult GetOptionsForSourcePath(string sourcePath)
 
             var normalizedPath = PathUtilities.NormalizeWithForwardSlash(sourcePath);
 
-            // The editorconfig paths are sorted from shortest to longest, so matches
-            // are resolved from most nested to least nested, where last setting wins
+            // The editorconfig paths are sorted from longest to shortest, so matches
+            // are resolved from least nested to most nested. This allows us to walk
+            // up the chain of editorconfig files until we find a root editorconfig
+            // or have processed them all.
```

Comment:
Instead of reversing the order, it feels like it might be a smaller change to clear the options when we hit a `root = true`.

Reply:
I like it. At first I thought this way would save processing, but since the config files are already parsed it wouldn't save much and it would be a little harder to follow. I'll update.